mobly/snippet/client_base.py

```diff
@@ -127,54 +116,42 @@ def start_server(self):
       errors.ServerStartError: when failed to start the snippet server.
     """
 
-    @contextlib.contextmanager
-    def _execute_one_stage(stage):
```

Comment:
do we not need the logging any more?
a little surprised to see the code structure changes with the name change.
(good to have simpler code for sure)

Reply:
When modifying the log statement, I feel that previous statement is redundant. So simplified the log statement and code at the same time.

Previously, the log was: 
```
[START_SERVER] Running the stage A.
[START_SERVER] Finished the stage A.
[START_SERVER] Running the stage B.
[START_SERVER] Finished the stage B.
[START_SERVER] Running the stage C.
[START_SERVER] Finished the stage C.
```

This PR modifies the log to be:
```
Preparing starting the snippet server for %s.
Starting the snippet server for %s.
Initializing a connection with the snippet server for %s.
```